src/bin/pgcopydb/schema.h

```diff
@@ -389,6 +389,11 @@ typedef struct DatabaseCatalog
 	Semaphore sema;
 } DatabaseCatalog;
 
+typedef struct TimelineHistoryContext
+{
+	DatabaseCatalog *source;
+} TimelineHistoryContext;
```

Comment:
I wonder why we don't just pass a `DatabaseCatalog *` argument around?

Reply:
Fair comment.

This struct used to have some more attributes that got removed. At some point I used to store the current timeline identifier for lookups in this context so that I can fetch the correct record from internal catalog. It no longer makes sense to introduce a new struct here at all.